### ISSUE TITLE
Doubts regarding Test Set and Hyper-Parameter Tuning

### ISSUE THREAD

**Issue by srikesh-07** [MIRALab-USTC/L2O-G2MILP#7]
Hello Authors @zijiegeng and @ziao-guo ,

First of all, I would like to thank the authors for their great research and publication.

1. In code **, We receive the test set but we don't use pre-process them and use them for the validation.**  Is there any specific reason we are using the training set for validation ?

2. I am currently doing the hyperparameter tuning. I can see the **modularity** of the graphs is low but the time-solving performance is increased higher. Can you please suggest me a way to increase the **modularity** score (or) what are the hyper-parameters that contributes to improve the **modularity** score?

I am looking for your inputs and response on above doubts.

Thank you

**Comment by zijiegeng** [comment]
Hi, Thank you for your question and sorry for the late response (as I was working on another job and missed the issue).
1. When determining the setting, I mainly followed some graph generation works in other areas such as drug design [1,2]. It appears a common setting to use the training set to evaluate the performance of a generative model and to see whether it can rerpoduce the distribution of the training set. It also makes sense to consider using an additional test set, and the results will not change a lot as the statistics in training and test set would be similar.
2. Currently no hyperparameter is designed to contral this specific score. I think it may help if you try adjusting the wights for different components in decoder (e.g., maybe a larger weight for the degree predictor). I think it will be a good job to find the relationship between some hyperparameters of components with the different final scores. If you are considering the trade-off between these metrics, personally I think the performance improvement resulted from data augmentation is a more important evaluation. BTW, if you see a different modularity score from previous results, that may be because @ziao-guo helped us fix a bug in computing the modularity: [(https://github.com/MIRALab-USTC/L2O-G2MILP/pull/1)]().

[1] [https://pubs.acs.org/doi/pdf/10.1021/acs.jcim.8b00839]()
[2] [https://openreview.net/pdf?id=Q_Jexl8-qDi]()
